core/src/main/java/jenkins/fingerprints/FingerprintStorage.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020, Sumit Sarin and Jenkins project contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.fingerprints;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+import java.io.IOException;
+
+import hudson.model.Fingerprint;
+import org.kohsuke.accmod.restrictions.Beta;
+import org.kohsuke.accmod.Restricted;
+
+/**
+ * Pluggable fingerprint storage API for fingerprints.
+ *
+ * @author Sumit Sarin
+ */
+@Restricted(Beta.class)
+public abstract class FingerprintStorage implements ExtensionPoint {
+
+    /**
+     * Returns the configured FingerprintStorage for the instance.
```

Comment:
This doesn't seem to be the configured one? afaict it's the first one and you've set the file one to have low priority so that an external one gets picked up first
```suggestion
     * Returns the first implementation of FingerprintStorage for the instance.
     * Implementation ordering is done via the ordinal attribute on the Extension
```

Reply:
Thanks for pointing that out @timja!